crates/bevy_ecs/src/system/query.rs

```diff
@@ -975,6 +975,35 @@ where
         self.state
             .is_empty(self.world, self.last_change_tick, self.change_tick)
     }
+
+    /// Returns `true` if the given [`Entity`] matches the query.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use bevy_ecs::prelude::*;
+    /// #
+    /// # #[derive(Component)]
+    /// # struct InRange;
+    /// #
+    /// # struct Target {
+    /// #     entity: Entity,
+    /// # }
+    /// #
+    /// fn targeting_system(in_range_query: Query<&InRange>, target: Res<Target>) {
+    ///     if in_range_query.contains(target.entity) {
+    ///         println!("Bam!")
+    ///     }
+    /// }
+    /// # targeting_system.system();
+    /// ```
+    #[inline]
+    pub fn contains(&self, entity: Entity) -> bool
```

Comment:
I'm wondering if its worth taking a chunk from `get_unchecked_manual` instead of using the whole function for performance reasons? 

for example, something like this in state.rs:
```rust
    pub fn contains<'w, 's>(&'s self, world: &'w World, entity: Entity) -> bool {
        let location = world.entities.get(entity);

        if let Some(location) = location {
            return self
                .matched_archetypes
                .contains(location.archetype_id.index());
        }
        false
    }

```

and then call `self.state.contains(self.world, entity)` inside query

Reply:
as @TheRawMeatball said earlier

> I think that a more optimised version is possible, but it would be challenging to review due to the amount of unsafety required.

Reply:
the above code doesn't call any unsafe functions and it would be easier enough to write a test to verify it works.

Reply:
example:
```rust
    #[test]
    fn query_contains_entity() {
        fn without_entity(entity: Res<Entity>, empty: Query<&A>) {
            assert!(!empty.contains(*entity));
        }

        fn with_entity(entity: Res<Entity>, not_empty: Query<&A>) {
            assert!(not_empty.contains(*entity));
        }
        let mut world = World::default();
        let entity = world.spawn().insert(B).id();
        world.insert_resource(entity);

        let mut without_entity = without_entity.system();
        without_entity.initialize(&mut world);
        without_entity.run((), &mut world);

        let entity = world.spawn().insert(A).id();
        world.insert_resource(entity);

        let mut with_entity = with_entity.system();
        with_entity.initialize(&mut world);
        with_entity.run((), &mut world);
    }
```

Reply:
I thought we would want to handle query filters though?